src/nodes/Collection.ts

```diff
@@ -207,4 +214,90 @@ export abstract class Collection extends NodeBase {
         )
     }
   }
+
+  /** Merge another collection into this */
+  merge(source: Collection|Document): void {
+    let sourceNode: Node;
+
+    if (source instanceof Document) {
+      if (isNode(source.contents)) {
+        sourceNode = source.contents
+      } else {
+        throw new Error(`Cannot merge Document content type "${typeof source.contents}" into Document`)
+      }
+    }  else if (isNode(source)) {
+      sourceNode = source
+    } else {
+      throw new Error(`Cannot merge node type "${typeof source}" into Document`)
+    }
```

Comment:
You're right, that would actually be nice. It wasn't my usecase, so I didn't think about it. I think just running createNode() should work.

> why is this checking isNode() rather than isCollection()
Again, a remnant. Initially I developed `merge()` as a Node Method, but then realized it is better suited on `Collection`. I'll fix that

Reply:
Okay, it's easy to just use `doc.createNode()` around the source to support `doc.merge({ foo: 'bar' })`. 

But inside a Collection I don't have access to the parent `Doc`'s `createNode()` method, right? I'd have to use the plain `createNode` method, so I have to identify the required Tag, and construct the node context. Is there any easier way?